### DISCUSSION TITLE
Color of graph based on other entity

### DISCUSSION THREAD

**Discussion by Lipown** [kalkih/mini-graph-card#842] [Ideas]
I am fetching spot price of electricity. My goal is to turn the parts of the graph green when I have low tariff and red if I have high tariff. The tariff is shown by binary sensor. Am I able to do this by mini graph card please?

**Comment by maranhaoBruno** [comment]
I am also trying to do this. I currently plot humidity in blue, but wish to have the entire graph change to red when another state variable changes from low to high. The determination of low and high is dynamic, and not a fixed value, hence why I don't want to utilize the color_thresholds feature. I have tried conditional statements within the line-color variable, but it has not worked for me. Instead I get a grey line. If anyone else knows how to solve this I would greatly appreciate it.

```- type: custom:mini-graph-card
    name: Humidity
    icon: mdi:water-percent
    hour24: true
    points_per_hour: 10
    entities:
      - sensor.master_bathroom_environment_humidity
    line_color: |-
      {% if ( is_state('input_boolean.master_bathroom_humid','on') ) %}
      red
      {% else %}
      var(--primary-color)
      {% endif %}

**Comment by geryduyck** [comment]
I have another use case.  I am using the card to display my temperatures, and would love to color the line based on the heat/cool/off modus :-) (so a red line when heating and a blue one when cooling, ...)

**Reply by ildar170975** [reply]
Currently there are these options:
1. Use color_thresholds with fixed “start points”. 
2. Use color_thresholds with dynamic “start points” (depending on other entities) - but you have to place mini-graph-card into a template-supporting wrapper-card like config-template-card or auto-entities.
3. Use card-mod with dynamic coloring depending on other entities - but only to make WHOLE graph colored with some color, not some part of a graph.

**Reply by geryduyck** [reply]
good idea using the car-mod for this, sometimes the simplest of solutions is right in front of you.
I currently was thinking about creating a card for each state, but was concerned about the performance, this will be more elegant! :-) 


I found this info on how to do it: https://community.home-assistant.io/t/lovelace-mini-graph-card/71385/2316 :-)

**Reply by BerkhanB** [reply]
I also wanted this. To achieve this, I tried to make a sensor that returns a set temperature value if the heating is on and unavailable (or none) if the heating is off. My idea was to display the sensor value on top of the set temperature so the set temperature would appear like changing colors. However, mini-graph-card just fills the unavailable data to make the graph more appealing. :(
